clang/test/CIR/CodeGen/switch.cpp

```diff
@@ -160,3 +159,62 @@ void sw7(int a) {
 // CHECK-NEXT: case (anyof, [3, 4, 5] : !s32i)  {
 // CHECK-NEXT:   cir.yield break
 // CHECK-NEXT: }
+
+void sw8(int a) {
+  switch (a)
+  {
+  case 3:
+    break;
+  case 4:
+  default:
+    break;
+  }
+}
+
+//CHECK:    cir.func @_Z3sw8i
+//CHECK:      case (equal, 3)
+//CHECK-NEXT:   cir.yield break
+//CHECK-NEXT:   },
+//CHECK-NEXT: case (default) {
+//CHECK-NEXT:   cir.yield break
+//CHECK-NEXT: }  
+
+void sw9(int a) {
+  switch (a)
+  {
+  case 3:
+    break;  
+  default:
+  case 4:
+    break;
+  }
+}
+
+//CHECK:    cir.func @_Z3sw9i
+//CHECK:      case (equal, 3)
+//CHECK-NEXT:   cir.yield break
+//CHECK-NEXT:   },
+//CHECK-NEXT: case (default) {
+//CHECK-NEXT:   cir.yield break
+//CHECK-NEXT: }
+
+void sw10(int a) {
+  switch (a)
+  {
+  case 3:
+    break;  
+  case 5:  
+  default:
+  case 4:
+    break;
+  }
+}
+
+//CHECK:    cir.func @_Z4sw10i
+//CHECK:      case (equal, 3)
+//CHECK-NEXT:   cir.yield break
+//CHECK-NEXT:   },
+//CHECK-NEXT: case (default) {
+//CHECK-NEXT:   cir.yield break
+//CHECK-NEXT: }
```

Comment:
@bcardosolopes does it matter if we prematurely remove irrelevant case statements?

Reply:
Nice catch. It does make a difference, it's fine as a extra opt step, but not right out of codegen, since this could prevent any sort of diagnostics we might wanna emit.

Reply:
This is similar to the problem solved by the folding cascading case logic, without of course shoving the `default` in the list because that doesn't make sense hehe.